### PR TITLE
Custom render context

### PR DESCRIPTION
**Pull request by usakhelo**
Hi Franz,
I've made two small changes:
1. Adds more complete RenderGlobalContext to make plugin compatible with extensions such VFB+
2. Implemented HasRequirement interface for max 2016 to make framebuffer 32-bit and to not show frambuffer when only project file is saved.

Thanks,
Sergo.

### CODE REVIEW

File: src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp [hunk 1]
```diff
@@ -730,11 +783,34 @@ void* AppleseedRenderer::GetInterface(ULONG id)
...
+bool AppleseedRenderer::HasRequirement(Requirement requirement)
+{
+    switch (requirement)
+    {
+      case kRequirement_NoVFB:
+      case kRequirement_DontSaveRenderOutput:
+        return m_settings.m_output_mode == RendererSettings::OutputMode::SaveProjectOnly;
+#if MAX_RELEASE < 19000
+      case kRequirement8_Wants32bitFPOutput: return true;
+#elif
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp]: You probably meant `#else` here.

File: src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp [hunk 2]
```diff
@@ -131,6 +131,59 @@ namespace
         { IDS_RENDERERPARAMS_LOG_OPEN_MODE_2,   L"Never" },
         { IDS_RENDERERPARAMS_LOG_OPEN_MODE_3,   L"On Error" }
     };
+
+    class AppleseedRenderContext
+        : public RenderGlobalContext
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp]: Half-indent.

File: src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp [hunk 3]
```diff
@@ -131,6 +131,59 @@ namespace
...
+    {
+    public:
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp]: Half-indent.

**Comment by usakhelo** [comment]
I think it's safe to merge this commit.

**Comment by usakhelo** [comment]
I also removed unnecessary nullptr's from previous PR (https://github.com/appleseedhq/appleseed-max/pull/192#discussion_r196300767)

**Pull request merged**